src/mirage_kv.mli

```diff
@@ -160,9 +160,9 @@ module type RO = sig
   (** [last_modified t k] is the last time the value bound to [k] in
      [t] has been modified.
 
-      When the value bound to [k] is a dictionary, the modification
-     time is the latest modification of all entries in that
-     dictionary. This behaviour is only one level deep and not recursive. *)
+      When the value bound to [k] is a dictionary, the implementation is free
+      to decide how to compute a last modified timestamp, or return [Error
+      (`Value_expected _)]. *)
```

Comment:
```suggestion
      (`Value_expected _)]. Earlier, the last-modified of a dictionary was defined as the latest modification
      in the dictionary (up to a depth of 1), but that didn't take value removal into account, and is neither what a
      Unix system provides. *)
```

Reply:
I am a bit hesitant about this. I feel it belongs in the CHANGES file. How long should we keep a description of the old behavior in the docs? WDYT?

Reply:
I agree, let's put the change in the changes, and keep the docstring brief to elaborate on the current interface.